src/Og.php

```diff
@@ -391,16 +391,14 @@ public static function getSelectionHandler(FieldDefinitionInterface $field_defin
       throw new \Exception("The field $field_name is not an audience field.");
     }
 
-    $options = NestedArray::mergeDeep([
+    $default_options = [
       'target_type' => $field_definition->getFieldStorageDefinition()->getSetting('target_type'),
       'handler' => $field_definition->getSetting('handler'),
-      'handler_settings' => [
-        'field_mode' => 'default',
-      ],
-    ], $options);
+      'field_mode' => 'default',
+    ] + $field_definition->getSetting('handler_settings');
```

Comment:
This retrieval of the `handler_settings` is still here to account for the fact that field definitions can be passed in that still use the deprecated handler settings. It's OK, the default implementation in core also still has this.

On a side note, it looks to me that this function is probably not really needed any more. The only thing that seems to make it different now from the standard implementation in `SelectionPluginManager::getSelectionHandler()` is that is allows to override the handler options, but it is only called once (in `og_entity_create_access()` and here the `$options` are not even passed. I will make a followup issue to deprecate and remove this.

Edit: created followup #580